flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlTicketResolver.java

```diff
@@ -0,0 +1,501 @@
+            case COMMAND_GET_TABLE_TYPES_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetTableTypes.class));
+            case COMMAND_GET_CATALOGS_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetCatalogs.class));
+            case COMMAND_GET_DB_SCHEMAS_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetDbSchemas.class));
+            case COMMAND_GET_TABLES_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetTables.class));
+            case COMMAND_GET_SQL_INFO_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetSqlInfo.class));
+            case COMMAND_GET_CROSS_REFERENCE_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetCrossReference.class));
+            case COMMAND_GET_EXPORTED_KEYS_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetExportedKeys.class));
+            case COMMAND_GET_IMPORTED_KEYS_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetImportedKeys.class));
+            case COMMAND_GET_PRIMARY_KEYS_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetPrimaryKeys.class));
+            case COMMAND_GET_XDBC_TYPE_INFO_TYPE_URL:
+                return execute(unpackOrThrow(any, CommandGetXdbcTypeInfo.class));
+        }
+        throw Exceptions.statusRuntimeException(Code.UNIMPLEMENTED,
+                String.format("FlightSQL command typeUrl '%s' is unimplemented", typeUrl));
+    }
+
+    private Table execute(SessionState sessionState, CommandStatementQuery query) {
+        if (query.hasTransactionId()) {
+            throw transactionIdsNotSupported();
+        }
+        return executeSqlQuery(sessionState, query.getQuery());
+    }
+
+    private Table execute(SessionState sessionState, CommandPreparedStatementQuery query) {
+        // Hack, we are just passing the SQL through the "handle"
+        return executeSqlQuery(sessionState, query.getPreparedStatementHandle().toStringUtf8());
+    }
+
+    private Table executeSqlQuery(SessionState sessionState, String sql) {
+        // See SQLTODO(catalog-reader-implementation)
+        // final QueryScope queryScope = sessionState.getExecutionContext().getQueryScope();
+        // Note: ScopeTicketResolver uses from ExecutionContext.getContext() instead of session...
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        // noinspection unchecked,rawtypes
+        final Map<String, Table> queryScopeTables =
+                (Map<String, Table>) (Map) queryScope.toMap(queryScope::unwrapObject, (n, t) -> t instanceof Table);
+        final TableSpec tableSpec = Sql.parseSql(sql, queryScopeTables, TicketTable::fromQueryScopeField, null);
+        // Note: this is doing io.deephaven.server.session.TicketResolver.Authorization.transform, but not
+        // io.deephaven.auth.ServiceAuthWiring
+        return tableSpec.logic()
```

Comment:
This recursive resolution is interesting. I'm curious what the benefits are using the TableCreator pattern vs. injecting the ScopeTicketResolver in as a dependency. I'm sure I'm missing an important benefit, but it feels a bit round-about when you only allow resolving scope variables.